services/automerge/automerge.go

```diff
@@ -303,4 +305,18 @@ func handlePullRequestAutoMerge(pullID int64, sha string) {
 		// on the pull request page. But this should not be finished in a bug fix PR which will be backport to release branch.
 		return
 	}
+
+	if pr.Flow == issues_model.PullRequestFlowGithub && scheduledPRM.DeleteBranchAfterMerge {
+		perm, err := access_model.GetUserRepoPermission(ctx, pr.HeadRepo, doer)
+		if err != nil {
+			log.Error("GetUserRepoPermission %-v: %v", pr.HeadRepo, err)
+			return
+		}
+
+		if perm.CanWrite(unit.TypeCode) { // default branch and branch protection will be checked in DeleteBranch
+			if err := repo_service.DeleteBranch(ctx, doer, pr.HeadRepo, headGitRepo, pr.HeadBranch); err != nil {
```

Comment:
Why not check the permission in `DeleteBranch`?

Reply:
e70e375